core/tests/performance_framework/selenium_driver.py

```diff
@@ -0,0 +1,174 @@
+    os.path.join(OPPIA_TOOLS_DIR, 'browsermob-proxy-0.7.1'),
+    os.path.join(OPPIA_TOOLS_DIR, 'selenium-2.53.2')
+]
+
+from browsermobproxy import Server
+from selenium import webdriver
+from xvfbwrapper import Xvfb
+
+CHROME_DRIVER_PATH = os.path.join(
+    '..', 'node_modules', 'protractor', 'selenium', 'chromedriver_2.21')
+CHROME_DRIVER = CHROME_DRIVER_PATH
+
+BROWSERMOB_PROXY_PATH = os.path.join(
+    '..', 'oppia_tools', 'browsermob-proxy-2.1.1', 'bin', 'browsermob-proxy')
+BROWSERMOB_PROXY = BROWSERMOB_PROXY_PATH
+
+
+class CaptureData(object):
+    """Class to fetch performance data"""
+
+    def __init__(self, browser='chrome'):
+        if browser in ['chrome', 'firefox']:
+            self.browser = browser
+        else:
+            # ToDo: Raise error here.
+            pass
+
+        if self.browser == 'chrome':
+            os.environ["webdriver.chrome.driver"] = CHROME_DRIVER
+
+        self.chrome_driver = CHROME_DRIVER
+        self.browser_mob = BROWSERMOB_PROXY
+        self.server = self.driver = self.proxy = None
+
+    def check_valid_url(self, page_url):
+        """Check if the url specified is valid."""
+        pass
+
+    def get_har(self, page_url):
+        """Method to get HTTP Archive (HAR) for a page_url
+
+        To get a HAR we require to use a proxy server.
+        """
+        with Xvfb() as xvfb:
+            self._setup_proxy()
+            self._setup_driver(use_proxy=True)
+
+            self.check_valid_url(page_url)
+
+            self.proxy.new_har(page_url, options={'captureHeaders': True})
+
+            self.driver.get(page_url)
+
+        result = self.proxy.har
+
+        self._clear_proxy()
+        self._clear_driver()
+
+        return result
+
+    def get_timings(self, page_url):
+        """Method to get Timings data for a page_url"""
+        with Xvfb() as xvfb:
+            self._setup_driver(use_proxy=False)
+
+            self.check_valid_url(page_url)
+
+            self.driver.get(page_url)
+
+            # Future ToDo: Firefox currently return function toJSON() for some keys.
+            # It return values for window.performance.timing, which is precisely
+            # what we need for now.
+            # To get entries for all the resources:
+            #   driver.execute_script("return window.performance.getEntries();")
+            # To get only the timing:
+            #   driver.execute_script("return window.performance.timing;")
+            performance = self.driver.execute_script("return window.performance")
+
+        self._clear_driver()
+
+        return performance
+
+    def _setup_proxy(self, downstream=None, upstream=None, latency=None):
+        self.server = Server(self.browser_mob)
+        self.server.start()
+        self.proxy = self.server.create_proxy()
+
+        # Set limits for the proxy server.
+        limits_specified = False
+        if downstream and upstream and latency:
+            limits_specified = True
+        if limits_specified:
```

Comment:
Rather than having ``limits_specified`` couldn't we simply do ``if downstream and upstream and latency`` here, instead?